src/amo/components/SectionLinks/index.js

```diff
@@ -2,40 +2,71 @@ import classNames from 'classnames';
 import React from 'react';
 import { compose } from 'redux';
 import { connect } from 'react-redux';
+import { withRouter } from 'react-router';
 
 import Link from 'amo/components/Link';
 import type { ViewContextType } from 'amo/reducers/viewContext';
+import { setClientApp } from 'core/actions';
 import {
   ADDON_TYPE_EXTENSION,
   ADDON_TYPE_THEME,
+  CLIENT_APP_ANDROID,
+  CLIENT_APP_FIREFOX,
   VIEW_CONTEXT_EXPLORE,
   VIEW_CONTEXT_HOME,
+  VIEW_CONTEXT_LANGUAGE_TOOLS,
 } from 'core/constants';
+import type { ApiStateType } from 'core/reducers/api';
 import translate from 'core/i18n/translate';
 import { visibleAddonType } from 'core/utils';
+import DropdownMenu from 'ui/components/DropdownMenu';
+import DropdownMenuItem from 'ui/components/DropdownMenuItem';
 
 import './styles.scss';
 
 
 type SectionLinksProps = {
   className?: string,
+  clientApp: string,
+  dispatch: Function,
   i18n: Object,
+  router: Object,
   viewContext: ViewContextType,
 }
 
 export class SectionLinksBase extends React.Component {
+  setClientApp = (event) => {
+    event.preventDefault();
+
+    const { dispatch, router } = this.props;
+
+    const clientApp = event.currentTarget.getAttribute('data-clientapp');
+
+    dispatch(setClientApp(clientApp));
+    router.push(event.currentTarget.getAttribute('href'));
+  }
+
   props: SectionLinksProps;
 
   render() {
-    const { className, i18n, viewContext } = this.props;
+    const { className, clientApp, i18n, viewContext } = this.props;
     const isExploring = [VIEW_CONTEXT_EXPLORE, VIEW_CONTEXT_HOME]
       .includes(viewContext);
 
+    let forBrowserNameText;
+    if (clientApp === CLIENT_APP_FIREFOX) {
+      forBrowserNameText = i18n.gettext('for Firefox');
+    } else if (clientApp === CLIENT_APP_ANDROID) {
+      forBrowserNameText = i18n.gettext('for Firefox for Android');
```

Comment:
Just to be sure: it is really "for Firefox for Android"? In both cases, it is for Firefox no? Why indicating that it is for Android when you probably run Firefox for Android on your Android device?

Reply:
I tested the feature locally and shouldn't this be "for Android"?

When you click on "Other Browser Sites > Add-ons for Firefox", you get:

<img width="219" alt="screen shot 2017-10-01 at 00 55 49" src="https://user-images.githubusercontent.com/217628/31050183-5bc025be-a643-11e7-8d7a-1559457403f4.png">

The first menu section header is named exactly like the link you've navigated to. However, if you click "Other Browser Sites > Add-ons for Android", then you get a slightly different section header:

<img width="227" alt="screen shot 2017-10-01 at 00 55 54" src="https://user-images.githubusercontent.com/217628/31050182-5bbe7f48-a643-11e7-8396-a8d759a64d7c.png">

Reply:
I wanted to be clear that we were targetting "Firefox for Android"... but that's a good point, it's confusing and inconsistent. Good catch; I'll switch it to just "for Android". Reads better, too. 👍